### DISCUSSION TITLE
[V5] Clean Arquitectura AdonisJs

### DISCUSSION THREAD

**Discussion by roliveira3891** [adonisjs/core#2467] [General]
hello! A doubt: could you give me a tip on how to structure adonisjs with Clean Architecture ?? thank you!

**Comment by cherifGsoul** [comment]
It may need more context to answer the question, but for now I would imagine putting all the custom code in a `lib` folder, like for Entities and Use Cases classes, interfaces adapters are also part of the lib folder.
Frameworks and drivers folder/layer is where interfaces adapters implementations will be, this part can be at adonis level or add them to the lib if you will, I would implement the interfaces that use Adonis components in the adonis app folder.

Dependency injection container is your friend for the clean architecture "Dependencies go inward" rule.

```
.
└── app-root/
    ├── app
    ├── ...
    ├── lib/
     │   ├── usecases
     │   ├── entities
     │    |── interfaces-adapters
    ├── ...
    └── resources
```

Another option a bit advanced and maybe complicated is to make monorepo where the custom code is organized in packages and you may create Adonis custom packages integrations and the app will use the custom adonis packages as dependencies.
